FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -436,11 +436,46 @@ variable [Algebra K (FiniteAdeleRing B L)]
 noncomputable def FiniteAdeleRing.baseChange : FiniteAdeleRing A K →ₐ[K] FiniteAdeleRing B L where
   toFun ak := ⟨ProdAdicCompletions.baseChange A K L B ak.1,
     (ProdAdicCompletions.baseChange_isFiniteAdele_iff A K L B ak).1 ak.2⟩
-  map_one' := sorry
-  map_mul' := sorry
-  map_zero' := sorry
-  map_add' := sorry
-  commutes' := sorry
+  map_one' := by
+    refine ext B L ?_
+    have h : (1 : FiniteAdeleRing A K) = (1 : ProdAdicCompletions A K) := rfl
+    have t : (1 : FiniteAdeleRing B L) = (1 : ProdAdicCompletions B L) := rfl
+    simp_rw [h,t]
+    simp only [map_one]
+  map_mul' := by
+    intro x y
+    have h : (x * y : FiniteAdeleRing A K) =
+      (x : ProdAdicCompletions A K) * (y : ProdAdicCompletions A K) := rfl
+    simp_rw [h]
+    simp only [map_mul]
+    rfl
+  map_zero' := by
+    refine ext B L ?_
+    have h : (0 : FiniteAdeleRing A K) = (0 : ProdAdicCompletions A K) := rfl
+    have t : (0 : FiniteAdeleRing B L) = (0 : ProdAdicCompletions B L) := rfl
+    simp_rw [h,t]
+    simp only [map_zero]
+  map_add' := by
+    intro x y
+    have h : (x+y : FiniteAdeleRing A K) =
+      (x : ProdAdicCompletions A K) + (y : ProdAdicCompletions A K) := rfl
+    simp_rw [h]
+    simp only [map_add]
+    rfl
+  commutes' r := by
+    refine ext B L ?_
+    have h : (((algebraMap K (FiniteAdeleRing A K)) r) : ProdAdicCompletions A K) =
+      (algebraMap K (ProdAdicCompletions A K)) r := rfl
+    simp_rw [h]
+    simp only [AlgHom.commutes]
+    have i : algebraMap K (FiniteAdeleRing B L) r =
+      algebraMap L (FiniteAdeleRing B L) (algebraMap K L r) := by
+      exact IsScalarTower.algebraMap_apply K L (FiniteAdeleRing B L) r
```

Comment:
```suggestion
        algebraMap L (FiniteAdeleRing B L) (algebraMap K L r) :=
      IsScalarTower.algebraMap_apply K L (FiniteAdeleRing B L) r
```
Who needs tactic mode? :-) `by` takes you from term mode to tactic mode, and `exact` takes you from tactic mode to term mode. So they cancel.